### PR TITLE
API-25378/ Update Docserver OAS

### PR DESCRIPTION
**Pull request by ediiotero**
Story: [API-25378](https://vajira.max.gov/browse/API-25378)

Generated an updated OAS for Docserver

- Updated summaries
- Added descriptions

### CODE REVIEW

**Review by mheusgit — COMMENTED** [review]
Without the docserver_oas_base.json I can't tell what new endpoints were added/removed.  You should have also had some new summary and descriptions tied to the new endpoints.

File: docs/docserver_oas_base.json [hunk 1]
```diff
@@ -12,17 +12,32 @@
...
         "summary": "View log settings",
-        "description": "Displays the set of active logging configurations",
         "consumes": [
           "application/json"
         ],
         "parameters": [],
-        "responses": {},
-        "tags": []
+        "responses": {
+          "200": {
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:20]: No responses should be in the base file.

File: docs/docserver_oas_base.json [hunk 2]
```diff
@@ -285,90 +322,46 @@
...
-                  "example": "Current Version"
-                }
-              }
-            }
+            "required": true
           }
         ],
         "responses": {},
         "tags": [],
-        "security": [
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:363]: Add the security back

File: docs/docserver_oas_base.json [hunk 3]
```diff
@@ -285,90 +322,46 @@
...
           {
-            "name": "majorVersion",
+            "name": "version",
             "in": "path",
-            "required": true,
-            "type": "integer",
-            "example": 1
-          },
-          {
-            "in": "body",
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:318]: Add the request body parameter back

File: docs/docserver_oas_base.json [hunk 4]
```diff
@@ -456,44 +451,28 @@
...
             "in": "path",
             "required": true
           },
           {
             "name": "oasIteration",
             "in": "path",
             "required": true
-          },
-          {
-            "in": "body",
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:469]: The parameter request bodies might have some updates but shouldn't be removed outright.

File: docs/docserver_oas_base.json [hunk 5]
```diff
@@ -456,44 +451,28 @@
...
-                  "type": "string",
-                  "example": "error"
-                }
-              }
-            }
           }
         ],
         "responses": {},
         "tags": [],
-        "security": [
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:486]: ditto

File: docs/docserver_oas_base.json [hunk 6]
```diff
@@ -525,24 +506,45 @@
...
           {
-            "name": "majorVersion",
+            "name": "version",
             "in": "path",
             "required": true
           }
         ],
         "responses": {},
         "tags": [],
-        "security": [
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:535]: ditto

File: docs/docserver_oas_base.json [hunk 7]
```diff
@@ -604,52 +610,555 @@
...
           },
           {
-            "name": "oasIteration",
+            "name": "version",
             "in": "path",
             "required": true
           }
         ],
         "responses": {},
         "tags": [],
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:643]: security property missing

File: docs/docserver_oas_base.json [hunk 8]
```diff
@@ -604,52 +610,555 @@
...
+            "name": "oasIteration",
+            "in": "path",
+            "required": true
+          }
+        ],
+        "responses": {},
+        "tags": [],
+        "description": "Runs background task to transform the original specification that is stored on DocServer.<br/><br/>  Transform only occurs based on API name, API version number, and OAS iteration ID.<br/>Change can be observed by viewing OAS list for a API version.",
+        "produces": [
+          "text/html"
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:708]: security property missing

File: docs/docserver_oas_base.json [hunk 9]
```diff
@@ -604,52 +610,555 @@
...
+    },
+    "/docserver/oas/cleanup": {
+      "delete": {
+        "summary": "Remove old OAS",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [],
+        "responses": {},
+        "tags": [],
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:763]: security property is missing

File: docs/docserver_oas_base.json [hunk 10]
```diff
@@ -604,52 +610,555 @@
...
+        ]
+      },
+      "post": {
+        "summary": "Create a new LOAST suite",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [],
+        "responses": {},
+        "tags": [],
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:787]: security and description properties are missing

File: docs/docserver_oas_base.json [hunk 11]
```diff
@@ -604,52 +610,555 @@
...
+        "description": "Remove OASs based on age and status"
+      }
+    },
+    "/docserver/loast/suites": {
+      "get": {
+        "summary": "View All LOAST suites",
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:769]: missing the description

File: docs/docserver_oas_base.json [hunk 12]
```diff
@@ -604,52 +610,555 @@
...
+        "produces": [
+          "text/plain",
+          "application/json",
+          "text/html"
+        ]
+      }
+    },
+    "/docserver/loast/suites/{name}": {
+      "get": {
+        "summary": "Get LOAST suite by name",
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:797]: description property is missing

File: docs/docserver_oas_base.json [hunk 13]
```diff
@@ -604,52 +610,555 @@
...
+        ],
+        "responses": {},
+        "tags": [],
+        "produces": [
+          "application/json"
+        ]
+      },
+      "put": {
+        "summary": "Update LOAST suite",
+        "consumes": [
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:816]: security and description

File: docs/docserver_oas_base.json [hunk 14]
```diff
@@ -604,52 +610,555 @@
...
+        "tags": [],
+        "produces": [
+          "text/plain",
+          "text/html",
+          "application/json"
+        ]
+      },
+      "delete": {
+        "summary": "Remove a LOAST suite",
+        "consumes": [
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:836]: security and description

File: docs/docserver_oas.json [hunk 15]
```diff
@@ -9047,351 +9398,729 @@
...
+        "tags": [],
+        "description": "Forwards request to appriate endpoint and its needed HTTP method using the 'task' parameter. <br/><br/>Endpoint intended only for external cronjobs to use since they have restrictions on their requests",
+        "produces": [
+          "text/plain",
+          "application/json",
+          "text/html"
+        ]
+      }
+    },
+    "/docserver/apis/{name}/versions/{majorVersion}": {
```
Comment (mheusgit) [inline docs/docserver_oas.json:9678]: There are still a couple of locations using "majorVersion" in the path

File: docs/docserver_oas_base.json [hunk 16]
```diff
@@ -525,24 +506,45 @@
...
+            "in": "path",
+            "required": true
+          },
+          {
+            "name": "version",
+            "in": "path",
+            "required": true
           }
+        ],
+        "responses": {},
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:537]: security property

File: docs/docserver_oas_base.json [hunk 17]
```diff
@@ -604,52 +610,555 @@
...
+    "/docserver/portal-store/metadata-all": {
+      "post": {
+        "summary": "Upload all metadata",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [],
+        "responses": {},
+        "tags": [],
+        "description": "Upload every APIs meta data to S3.<br/><br/>Data is stored on S3 under '{API name}/metadata.json'",
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:989]: security is missing

File: docs/docserver_oas_base.json [hunk 18]
```diff
@@ -604,52 +610,555 @@
...
+        "parameters": [
+          {
+            "type": "string",
+            "name": "task",
+            "in": "query",
+            "example": "/oas/refresh"
+          }
+        ],
+        "responses": {},
+        "tags": [],
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:1040]: This is the only "GET" method that requires a security property

File: docs/docserver_oas_base.json [hunk 19]
```diff
@@ -659,66 +1168,89 @@
...
+          }
+        ],
+        "responses": {},
+        "tags": []
+      },
+      "patch": {
+        "summary": "Change OAS iteration status",
+        "description": "Change OAS iteration to 'inactive', 'error', or 'review' status.  Enforces policy of no 'inactive' iterations after 'active' iteration, and no 'review' iterations before active iteration.<br/><br/> Status change is based on API name, API version number, and OAS iteration ID",
         "consumes": [
           "application/json"
```
Comment (mheusgit) [inline docs/docserver_oas_base.json:1225]: security

Comment (mheusgit) [reply docs/docserver_oas_base.json:1340]: Fired up this OAS in my local OAS generator and see patch "[/docserver/apis/{name}/versions/{version}/oas/{oasIteration}](http://localhost:8080/docserver/api-docs/#/default/patch_docserver_apis__name__versions__version__oas__oasIteration_)" is still missing the security

Comment (mheusgit) [reply docs/docserver_oas_base.json:1340]: 14 endpoints still use "majorVersion" in the path

**Comment by mheusgit** [comment]
With 55 endpoints the OAS is getting to the point we need the labeling added. Gonna toss it in a a bonus

File: docs/docserver_oas_base.json (hunk 19 above)
Comment (mheusgit) [reply docs/docserver_oas_base.json]: Duplication has been correctly.  I had to add the patch security to get the ball rolling

File: docs/docserver_oas_base.json [hunk 20]
```diff
@@ -525,24 +636,55 @@
...
         "security": [
           {
             "basicAuth": []
           }
+        ],
+        "produces": [
+          "text/html"
+        ]
+      },
+      "delete": {
```
Comment (christineyost) [inline docs/docserver_oas_base.json:655]: description is missing

**Comment by dsteeve** [comment]
I am getting errors when I clicked thru some of the endpoints to view them via swagger UI. 
This is probably related to why the 200 and 201 responses are empty.

Screenshot 2023-04-26 at 9 31 14 AM

File: docs/docserver_oas_base.json [hunk 21]
```diff
@@ -795,150 +1222,20 @@
...
-            }
-          }
-        ],
-        "responses": {},
-        "tags": [],
-        "security": [
-          {
-            "basicAuth": []
-          }
+        "description": "Forwards request to appriate endpoint and its needed HTTP method using the 'task' parameter. <br/><br/>Endpoint intended only for external cronjobs to use since they have restrictions on their requests",
```
Comment (dsteeve) [inline docs/docserver_oas_base.json:1226]: typo : "appriate" should be "appropriate"

File: docs/docserver_oas_base.json [hunk 22]
```diff
@@ -639,30 +874,204 @@
...
+      }
+    },
+    "/docserver/oas/cleanup": {
+      "delete": {
+        "summary": "Remove old OAS",
         "consumes": [
           "application/json"
         ],
         "parameters": [],
         "responses": {},
```
Comment (dsteeve) [inline docs/docserver_oas_base.json:912]: I think this takes in query params 
export interface OASDeleteQuery {
  oasStatus: StatusEnum;
  olderThan: number;
}
and olderThan is default to 120 days if none is given

l think oasStatus is required (?)

File: docs/docserver_oas_base.json [hunk 23]
```diff
@@ -795,150 +1222,20 @@
...
-            "in": "path",
-            "required": true
-          },
-          {
-            "name": "majorVersion",
-            "in": "path",
-            "required": true
-          },
-          {
-            "type": "string",
```
Comment (dsteeve) [inline docs/docserver_oas_base.json:820]: I think page, page_size, and total_required need to still be here

**Pull request merged**